enhancements/deprecated-bundles.md

```diff
@@ -0,0 +1,131 @@
+- [ ] Test plan is defined
+
+## Summary
+
+OPM will support marking versions as deprecated, truncating the update graph of a given package, and appending to the tail of channel. Deprecated versions will not be installable. If a deprecated version is installed prior to upgrade appropriate alerts will be fired on cluster. 
+
+## Motivation
+
+The default set of Operators provided on OCP clusters will soon (4.6+) be delivered through indexes and each OCP minor version will have a unique set of index images. Today, packages in one index will be duplicated to the next minor version index when it is available. This means operator owners releasing through the redhat, certified, or community pipelines will now need a way to specify which versions of their operator are incompatible or unsupported in the next minor version index.
+
+### Terminology
+
+#### Deprecated bundle
+
+A deprecated bundle is an upgradable but not installable bundle in the update graph of an operator.
+
+#### Terminated bundle
+
+A terminated bundle is neither upgradable nor installable.
+
+### Goals
+
+- A deprecated bundle in the update graph and any of its replacements (terminated bundles) will not be installed by OLM
+- Operator Owners don’t need to release a new version in order to specify which bundle is deprecated
+- The index add command is enhanced to un-do this termination if needed
+- The deprecation is specified after promotion (not during) and before publishing of the index image
+- Alerts are fired on cluster when a deprecated version is installed
+
+### Non-Goals
+
+- OLM supports upgrading from a terminated bundle
+- Alerts are fired if terminated bundles exist on cluster
+- OLM prevents cluster upgrades based on whether versions are terminated in the upgraded cluster’s index
+
+## Proposal
+
+We propose to add a command to `opm index` called `deprecate` which takes in a set of bundle images, an index, and all the same arguments as `opm index add`. The command modifies the index by removing all versions that come before the specified bundle image in the update graph (that is, until the tail of each channel). If this operation removes the head of a channel, then the channel is removed. The removal of bundles from the update graph will not modify the update graph after the deprecated version (from the deprecated channel to the head of the channel).
+
+Versions that were removed from the index will not receive updates if subscriptions still exist for them on cluster. Ideally, appropriate communication from the operator maintainer will mean that cluster admins will upgrade to the latest supported version prior to cluster upgrade. In the event that the cluster is upgraded and `Subscriptions` to terminated bundles exist on cluster, the cluster admin can manually trigger an upgrade by either.
+
+- editing the `Subscription`'s `startingCSV` field (and/or `channel` field if the channel was removed) to point to an available / installable `csv`
```

Comment:
@bparees This is the section that follows up on the conversation we had offline. 

Two Questions:
1. Is this more clear what happens when a subscription is stranded? 
2. Does this UX work, or do you have reservations about it?

The alternative is that we attempt to also keep every node and some as not installable. I think that over time this becomes unwieldy and we will want to have this mechanism anyway?

Reply:
/cc @ecordell @kevinrizza

Reply:
After re-reading this, we need to remove the `staringCSV` field update causing an upgrade. We should instead make it `source` field update or `channel` update because we will need to change the source to the custom catalog.